api/tests/provider_api_tests.py

```diff
@@ -13,42 +13,63 @@ class BaseAPITest(APITestCase):
     def setUp(self):
         self.request_factory = RequestFactory()
         self.view_set = ProviderListViewSet
-        self.base_url = '/api/providers/'
 
-    def get_request(self, id=None):
-        request = self.request_factory.get(f'{self.base_url}{id}')
+    def get_request_factory(self, url, average_ratings={}, id=None):
```

Comment:
You can make us of kwargs here https://www.geeksforgeeks.org/args-kwargs-python/ .
As all the subclasses of this have an url set, we can use `self.url` and omit passing it as an argument.